api/api/utils/generate_teams_data_loader.py

```diff
@@ -1,40 +1,76 @@
-from typing import Dict, Any, List
-from dataclasses import dataclass, field
+from typing import Dict, Any, List, Optional
+from dataclasses import dataclass
+
+from schema import Schema
 
 from api.ai.new.interfaces.algorithm_options import AlgorithmOptions
 from api.ai.new.interfaces.team_generation_options import TeamGenerationOptions
 from api.models.enums import AlgorithmType
-from api.models.project import Project
 from api.models.student import Student
+from api.api.utils.algorithm_option import get_algorithm_option_class
 
 
 @dataclass
 class GenerateTeamsInputData:
     students: List[Student]
     algorithm_type: AlgorithmType
-    algorithmOptions: AlgorithmOptions
+    algorithm_options: AlgorithmOptions
     team_generation_options: TeamGenerationOptions
-    projects: List[Project] = field(default_factory=list)
 
 
 class GenerateTeamsDataLoader:
     def __init__(self, validated_data: Dict[str, Any]):
         self.data = validated_data
 
     def load(self) -> GenerateTeamsInputData:
-        pass
+        return GenerateTeamsInputData(
+            students=self._get_students(),
+            team_generation_options=self._get_team_generation_options(),
+            algorithm_type=self._get_algorithm_type(),
+            algorithm_options=self._get_algorithm_options(),
+        )
 
     def _get_algorithm_type(self) -> AlgorithmType:
-        pass
+        algorithm_options = self.data.get("algorithm_options")
+        algorithm_type = algorithm_options.get("algorithm_type")
+        return AlgorithmType(algorithm_type)
 
     def _get_students(self) -> List[Student]:
-        pass
-
-    def _get_projects(self) -> List[Project]:
-        pass
+        students = self.data.get("students")
+        return [
+            Student(
+                _id=student.get("id"),
+                name=student.get("name"),
+                relationships=student.get("relationships"),
```

Comment:
like why not just convert this dict to be `Dict[str, Relationship]` now and then we know the relationship values are always of the enum type going forward?

Reply:
The loader's job is to make sure the data is consistent with the rest of the module